engine/util/rng.hpp

```diff
@@ -65,33 +68,45 @@ struct basic_rng_t
   /// Uniform distribution in the range [min..max)
   double range( double min, double max );
 
-  /// Uniform distribution in the range [min..max)
-  template <typename T, typename = std::enable_if_t<std::is_integral<T>::value>>
-  T range( T min, T max ) {
-    return static_cast<T>(range(static_cast<double>(min), static_cast<double>(max)));
-  }
-
-  /// Uniform distribution in the range [0..max)
-  template <typename T, typename = std::enable_if_t<std::is_integral<T>::value>>
-  T range( T max ) {
-    return range<T>( T{}, max );
-  }
+  /// Uniform distribution in the range [0.0..max)
+  double range( double max );
 
   /// Gaussian Distribution
-  double gauss( double mean, double stddev, bool truncate_low_end = false );
+  double gauss( double mean, double stddev );
+
+  // Truncated Gaussian Distribution
+  double gauss_ab( double mean, double stddev, double min, double max );
+  double gauss_a( double mean, double stddev, double min );
+  double gauss_b( double mean, double stddev, double max );
 
   /// Exponential Distribution
   double exponential( double nu );
 
   /// Exponentially Modified Gaussian Distribution
   double exgauss( double gauss_mean, double gauss_stddev, double exp_nu );
 
+  template <typename T, typename std::enable_if_t<!std::numeric_limits<T>::is_signed>>
+  T gauss( T mean, T stddev )
+  {
+    return static_cast<T>( gauss_a( static_cast<double>( mean ), static_cast<double>( stddev ), 0.0 ) );
+  }
+
+  template <typename T, typename std::enable_if_t<!std::numeric_limits<T>::is_signed>>
+  T exgauss( T gauss_mean, T gauss_stddev, T exp_nu )
+  {
+    return static_cast<T>( gauss_a( static_cast<double>( gauss_mean ), static_cast<double>( gauss_stddev ), 0.0 ) +
+                           exponential( static_cast<double>( exp_nu ) ) );
+  }
```

Comment:
This won't work. The template line needs to be:
```cpp
template <typename T, typename = std::enable_if_t<!std::numeric_limits<T>::is_signed>>
```
Note the additional equals sign.

Reply:
Thanks, I was using partial specializations previously (which ended out identical in template to what I had, albeit with a forward declaration above) and had not realized that they're syntactically different.